xclim/utils.py

```diff
@@ -62,6 +64,275 @@
+
+        ds['time'] = xr.decode_cf(ds).time
+        ds['time'].values = pd.to_datetime({'year': ds.time.dt.year, 'month': ds.time.dt.month, 'day': ds.time.dt.day})
+
+        ds = ds.where((ds.time >= start1) & (ds.time <= end1), drop=True)
+
+        ds1.append(ds.drop('time'))
+    print('concatenating files : adding dimension ', dim, )
+    ens = xr.concat(ds1, dim=dim)
+    # assign time coords
+    ens = ens.assign_coords(time=ds.time.values)
+    return ens
+
+
+def ensemble_percentiles(ens, values=[10, 50, 90], time_block=None):
+    """Calculate ensemble statistics between a results from an ensemble of climate simulations
+
+    Returns a dataset containing ensemble statistics for input climate simulations.
+    Alternatively calculate ensemble percentiles (default) or ensemble mean and standard deviation
+
+
+    Parameters
+    ----------
+
+    ens : Ensemble dataset (see xclim.utils.create_ensemble)
+
+    values (optional) : list of integers : percentile values to calculate (default : 10,50, 90)
+
+    time_block (optional) : integer : for large ensembles iteratively calculate percentiles
+    in time-step blocks (n==time_block).  If not defined the function tries to estimate an appropriate value
+
+
+    Returns
+    -------
+    xarray dataset with containing data variables of requested ensemble statistics
+
+    Examples
+    --------
+
+    >>> from xclim import utils
+    >>> import glob
+    >>> ncfiles = glob.glob('/*tas*.nc')
+    Create ensemble dataset
+    >>> ens = utils.create_ensemble(ncfiles)
+    Calculate default ensemble percentiles
+    >>> ens_percs = utils.ensemble_statistics(ens)
+    >>> print(ens_percs['tas_p10'])
+    Calculate non-default percentiles (25th and 75th)
+    >>> ens_percs = utils.ensemble_statistics(ens, values=[25,75])
+    >>> print(ens_percs['tas_p25'])
+    Calculate by time blocks (n=10) if ensemble size is too large to load in memory
+    >>> ens_percs = utils.ensemble_statistics(ens, time_block=10)
+    >>> print(ens_percs['tas_p25'])
+
+    """
+
+    dsOut = ens.drop(ens.data_vars)
+    dims = list(ens.dims)
+    for v in ens.data_vars:
+        # Percentile calculation requires load to memory : automate size for large ensemble objects
+        if not time_block:
+            time_block = round(2E8 / (ens[v].size / ens[v].shape[dims.index('time')]), -1)  # 2E8
+
+        if time_block > len(ens[v].time):
+            Out = calc_percentiles_simple(ens, v, values)
+
+        else:
+            # loop through blocks
+            Warning('large ensemble size detected : statistics will be calculated in blocks of ', int(time_block),
+                    ' time-steps')
+            Out = calc_percentiles_blocks(ens, v, values, time_block)
+        for vv in Out.data_vars:
+            dsOut[vv] = Out[vv]
+    return dsOut
+
+
+def calc_percentiles_simple(ens, v, values):
+    dsOut = ens.drop(ens.data_vars)
+    dims = list(ens[v].dims)
+    outdims = [x for x in dims if 'sim' not in x]
```

Comment:
You are using sim as the dimension name, but this can be changed by the user in the create_ensemble function. 

One option is to pass an IndexVariable to concat as the dim (instead of just a string) and make sure that it has attribute axis='E'. You could then find the dimension with the axis='E' attribute to identify the ensemble dimension.